microsoft-edge/webview2/concepts/developer-guide.md

```diff
@@ -0,0 +1,80 @@
+ 
+To solve this situation, you can test for the availability of the APIs in the installed runtime, before making a call to the API. To test for API availability in the installed runtime, use:
+* The `queryinterface` in C/C++. For more information, navigate to <!--LINK-->
+* A try/catch block in .Net or WinUI, as displayed in the following code sample. 
+
+```csharp
+    /// Add code snippet that demonstrates how to use the try/catch block
+```
+    
+For more information, navigate to [Determine WebView2 Runtime requirement][Webview2ConceptsVersioningDetermineWebview2RuntimeRequirement].  
+
+## Update the Fixed Version Runtime  
+
+Depending on your environment, work stoppage due to applications going offline may significantly impact your business.  For example, certain financial or medical applications may have high availability requirements. In these scenarios, you may be required to use the Fixed Version Runtime to control updates to your application. If you use the Fixed Version Runtime, ensure you update your runtime regularly to reduce any potential security risk. When using 3rd-party content in Webview2 applications, always consider the content untrusted.  For more information, navigate to [Fixed Version distribution mode][Webview2ConceptsDistributionFixedVersionDistributionMode].  
+
+## Manage new versions  
+
+Whenever a new version of the Evergreen WebView2 Runtime has been downloaded to the device, running WebView2 applications continue to hold onto the browser process of the previous runtime. To use the new version of the runtime, you'll need to restart or quit the running application. The benefits of this model are any running WebView2 applications won't break and there's no additional memory allocated to a new browser process. A disadvantage of this model is there are additional binaries installed on the disk that won't get cleaned up.
+
+To take action when a new version is available, such as notifying the user to restart the application, you can use the `new version available` event in your .Net and C++ code. If your code handles restarting the application, consider saving the user state before the WebView2 application exits. The following code sample shows how you can use the `new version available` event.
+
+```cpp
+    /// Add code snippet that demonstrates how to use the new version available event
+```
+
+```csharp
+    /// Add code snippet that demonstrates how to use the new version available event
+
+```
+
+## Manage the lifetime of the user data folder 
+When your WebView2 application is installed, a user data folder is created, and contains user data such as cookies, credentials, permissions, and so on.  After the folder is created, your app is responsible for managing and cleaning up the user data folder.  For more information, navigate to [Managing the User Data Folder][Webview2ConceptsUserdatafolder].  
```

Comment:
> When your WebView2 application is installed, a user data folder is created, and contains user data such as cookies, credentials, permissions, and so on.  After the folder is created, your app is responsible for managing and cleaning up the user data folder.

WebView2 creates a user data folder to store data such as cookies, credentials, permissions, and so on. After the folder is created, your app is responsible for managing the lifetime of user data folder, such as clean-up upon app uninstallation.